tickets/src/components/content/EventContent.js

```diff
@@ -1,68 +1,172 @@
-import React, { Fragment } from 'react'
+import React, { Component } from 'react'
 import styled from 'styled-components'
+import { connect } from 'react-redux'
+import Head from 'next/head'
 import PropTypes from 'prop-types'
 
 import { Fieldset, Field, Label } from './CreateContent'
-import { MONTH_NAMES } from '../../constants'
+import { MONTH_NAMES, pageTitle, TRANSACTION_TYPES } from '../../constants'
 import UnlockPropTypes from '../../propTypes'
 import BalanceProvider from '../helpers/BalanceProvider'
+import { lockRoute } from '../../utils/routes'
+import BrowserOnly from '../helpers/BrowserOnly'
+import Layout from '../interface/Layout'
+import GlobalErrorConsumer from '../interface/GlobalErrorConsumer'
+import { purchaseKey } from '../../actions/key'
+import { loadEvent } from '../../actions/ticket'
+import PayButton from './purchase/PayButton'
 
-export const EventContent = ({ name, description, location, date, lock }) => {
-  let dateString =
-    MONTH_NAMES[date.getMonth()] +
-    ' ' +
-    date.getDate() +
-    ', ' +
-    date.getFullYear()
-
-  return (
-    <Fragment>
-      <Title>{name}</Title>
-      <DetailsFieldset>
-        <Field>
-          <Label>Ticket Price</Label>
-          <BalanceProvider
-            amount={lock.keyPrice}
-            render={(ethWithPresentation, convertedUSDValue) => (
-              <Price>
-                <Eth>{ethWithPresentation} ETH</Eth>
-                <Fiat>${convertedUSDValue}</Fiat>
-              </Price>
-            )}
-          />
-        </Field>
-        <PayButton>Pay &amp; Register for This Event</PayButton>
-      </DetailsFieldset>
-      <DetailsFieldset>
-        <DetailsField>
-          <DisplayDate>{dateString}</DisplayDate>
-          <Description>{description}</Description>
-        </DetailsField>
-        <Field>
-          <Label>Location</Label>
-          <Description>{location}</Description>
-        </Field>
-      </DetailsFieldset>
-    </Fragment>
-  )
+export class EventContent extends Component {
+  constructor(props) {
+    super(props)
+
+    let { event } = this.props
+    if (!event) event = {}
+
+    this.state = {
+      event: event,
+    }
+
+    this.loadEvent.bind(this)
+  }
+
+  componentDidUpdate() {
+    const { lock } = this.props
+    const { event } = this.state
+    if (lock.address && !event.lockAddress) loadEvent(lock.address)
```

Comment:
I think `loadEvent` needs to go in the `mapDispatchToProps` so that the action can get into the middlewares